css/src/components/progress-bar.scss

```diff
@@ -0,0 +1,92 @@
+$progress-bar-height-medium: 0.125rem !default;
+$progress-bar-height-large: 0.25rem !default;
+
+$progress-bar-container-background-color: $secondary-background !default;
+$progress-bar-background-color: $primary !default;
+$progress-bar-success-background-color: $success !default;
+$progress-bar-warning-background-color: $warning !default;
+$progress-bar-danger-background-color: $danger !default;
+$progress-bar-border-radius-rounded: $border-radius-rounded !default;
+
+$progress-bar-animation-duration: 2s !default;
+$progress-bar-transition-duration: 0.3s !default;
+
+.progress-bar-container {
+	position: relative;
+	width: 100%;
+	height: $progress-bar-height-medium;
+	background-color: $progress-bar-container-background-color;
+	overflow: hidden;
+
+	@include forced-colors {
+		background-color: CanvasText !important;
+	}
+
+	.theme-high-contrast & {
+		background-color: $secondary-background-glow-high-contrast;
+	}
+
+	&.progress-bar-md {
+		height: $progress-bar-height-medium;
+	}
+
+	&.progress-bar-lg {
+		height: $progress-bar-height-large;
+	}
+}
+
+@mixin indeterminate-gradient($color) {
+	position: absolute;
+	width: 100%;
+	animation: progress-bar-indeterminate-animation $progress-bar-animation-duration infinite linear;
+	background: linear-gradient(to right, transparent 0%, $color 50%, transparent 100%);
+
+	@include prefers-reduced-motion {
+		animation-duration: 0s;
+	}
+
+	&:dir(rtl) {
+		background: linear-gradient(to left, transparent 0%, $color 50%, transparent 100%);
+	}
+
+	inset-block-start: 0;
+	inset-inline-start: 0;
+}
+
+.progress-bar {
+	height: 100%;
+	transition: width $progress-bar-transition-duration ease;
+	background-color: $progress-bar-background-color;
+
+	@include forced-colors {
+		background-color: Highlight !important;
```

Comment:
**Note for other reviewers:** The use of the `Highlight` system color for the completed portion, and `CanvasText` for the uncompleted portion of the track is copied from Fluent.